### ISSUE TITLE
[Block] Classic: Media with captions shows caption HTML around image in editor

### ISSUE THREAD

**Issue by cuemarie** [WordPress/gutenberg#53698]
### Description

Images containing captions added via the Classic Block have begun showing `[caption] [/caption]` html around the image and caption, only in the block editor. 

On the live site, and in the Classic Editor pop-up modal, the caption appears as expected.

### Step-by-step reproduction instructions

1. Create a new post
2. Add the Classic block
3. Within the classic block, add an image with a caption set, and save

### Screenshots, screen recording, code snippet

Pop-Up Classic Editor:

![Markup on 2023-08-15 at 11:45:21](https://github.com/WordPress/gutenberg/assets/27249804/1794352f-d6ff-4c0c-a91d-241e2ab2b228)

⚠️ Block Editor:

![Markup on 2023-08-15 at 11:45:31](https://github.com/WordPress/gutenberg/assets/27249804/ef3826dd-7086-4713-b8de-b8e7d23df773)

Live Site:

![Markup on 2023-08-15 at 11:45:56](https://github.com/WordPress/gutenberg/assets/27249804/48fb164c-c071-4605-89bd-ef34e0a0f92c)

### Environment info

- WordPress 6.3
- Gutenberg v16.4.0 _(rolling back to 16.3.0 does not resolve, caption issue present there as well)_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jordesign** [comment]
Tested this and confirmed.

**Comment by Mustafabharmal** [comment]
Initial approach is to align the Classic block preview with frontend behavior by rendering content via a server-side callback instead of raw HTML. This should ensure shortcodes (like `[caption]`) are processed correctly in the editor preview as well.

**Comment by Mamaduka** [comment]
IIRC, the old editor had special handling for some shortcodes - `cation`, `gallery`, etc. I don't think using SSR to render every shortcode is the right approach here.

Why?

* Most shortcodes require dynamic data, which can be different between the front-end and dashboard.
* Classic block was meant to be a temporary solution while migrating to blocks. It's not a one-to-one replacement of Classic Editor.
* There's an ongoing effort to deprecate the Classic block. See https://github.com/WordPress/gutenberg/pull/77838.

cc @tyxla

**Comment by tyxla** [comment]
Yeah, we're not planning to invest in making the Classic block better. 

As @Mamaduka said, it was intended to be a temporary solution while migrating to blocks. On the contrary, we are seeking to make the Classic block less tempting to use, and as a continuation of the work in https://github.com/WordPress/gutenberg/pull/77838, we'll explore how to make it more compelling to move away from it.

cc @yuliyan, who has been helping with this effort.

**Comment by Mamaduka** [comment]
> We'll explore how to make it more compelling to move away from it.

100% - improving Classic to Block coversion, would help with this effort.

Let's close this as won't fix.
